sklearn/_loss/loss.py

```diff
@@ -770,6 +771,52 @@ def constant_to_optimal_zero(self, y_true, sample_weight=None):
             return term
 
 
+class HalfTweedieLossIdentity(BaseLoss):
```

Comment:
My curiosity: when does it make sense to use identity link with power != 0 ?

Reply:
Years ago, I thought it a good idea. Meanwhile, I don't think it's is useful. Therefore, I opened #19086 without much response.

Reply:
> My curiosity: when does it make sense to use identity link with power != 0 ?

I have a problem where I know the expectation y' follows the linear model y' = w x. My measurements, y, have poisson errors. 

(The specific problem involves analysis of radiation measurements. The expectation is linear with the amount of source; the measurements are poisson distributed).

Using a log link function is just not the right description of my problem. Yes, the whole thing breaks down when evaluating negative values of w, but it seems much better to offer a constraint to avoid ever evaluating negative values of w rather than exclude the situations where you have an actual linear relationship with poisson measurements.